### PR TITLE
Remove SpanData per the RFC.

### PR DESCRIPTION
**Pull request by bogdandrutu**
This PR does not add the new options to set timestamps for the start/end, just for the moment removes SpanData.

* Timestamp was moved in it's own class in preparation to be added as an option to start/end.
* Default implementations for Link/Event were moved in util.

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/trace/Timestamp.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2017, OpenTelemetry Authors
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/trace/Timestamp.java]: s/2017/2019

File: api/src/main/java/io/opentelemetry/trace/util/Events.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.internal.Utils;
+import io.opentelemetry.trace.AttributeValue;
+import io.opentelemetry.trace.Event;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.concurrent.Immutable;
+
+public class Events {
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/trace/util/Events.java]: Javadoc?

File: api/src/main/java/io/opentelemetry/trace/util/Links.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.trace.AttributeValue;
+import io.opentelemetry.trace.Link;
+import io.opentelemetry.trace.SpanContext;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.concurrent.Immutable;
+
+public final class Links {
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/trace/util/Links.java]: ditto

File: api/src/test/java/io/opentelemetry/trace/util/EventsTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import io.opentelemetry.trace.AttributeValue;
+import io.opentelemetry.trace.Event;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link Links}. */
```
Comment (songy23) [inline api/src/test/java/io/opentelemetry/trace/util/EventsTest.java]: s/Links/Events

File: api/src/main/java/io/opentelemetry/trace/Timestamp.java (hunk 1 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/trace/Timestamp.java]: done.

File: api/src/main/java/io/opentelemetry/trace/util/Events.java (hunk 2 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/trace/util/Events.java]: done.

File: api/src/main/java/io/opentelemetry/trace/util/Links.java (hunk 3 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/trace/util/Links.java]: done.

File: api/src/test/java/io/opentelemetry/trace/util/EventsTest.java (hunk 4 above)
Comment (bogdandrutu) [reply api/src/test/java/io/opentelemetry/trace/util/EventsTest.java]: done.

**Comment by pavolloffay** [comment]
There are a couple of references to `SpanData` in comments or method names

```
sdk/src/main/java/io/opentelemetry/sdk/trace/TraceProtoUtils.java:40:// Utilities to convert Span SDK and SpanData to trace protos.
sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSampledSpansProcessor.java:279:    // Exports the list of SpanData to all the ServiceHandlers.
sdk/src/main/java/io/opentelemetry/sdk/trace/export/InMemorySpanExporter.java:40: *   public void getFinishedSpanData() {
sdk/src/main/java/io/opentelemetry/sdk/trace/export/InMemorySpanExporter.java:51:  private final List<Span> finishedSpanDataItems = new ArrayList<>();
sdk/src/main/java/io/opentelemetry/sdk/trace/export/InMemorySpanExporter.java:71:      return Collections.unmodifiableList(new ArrayList<>(finishedSpanDataItems));
sdk/src/main/java/io/opentelemetry/sdk/trace/export/InMemorySpanExporter.java:82:      finishedSpanDataItems.clear();
sdk/src/main/java/io/opentelemetry/sdk/trace/export/InMemorySpanExporter.java:92:      finishedSpanDataItems.addAll(spans);
sdk/src/main/java/io/opentelemetry/sdk/trace/export/InMemorySpanExporter.java:100:      finishedSpanDataItems.clear();
sdk/src/test/java/io/opentelemetry/sdk/trace/export/InMemorySpanExporterTest.java:43:  public void getFinishedSpanDataItems() {
```

**Comment by bogdandrutu** [comment]
@pavolloffay done.

**Pull request merged**